runtime/caml/sync.h

```diff
@@ -76,15 +76,18 @@ Caml_inline int sync_condvar_wait(sync_condvar c, sync_mutex m)
 
 Caml_inline void sync_check_error(int retcode, char * msg)
 {
-  char * err;
+  char err[1024];
   int errlen, msglen;
   value str;
 
   if (retcode == 0) return;
   if (retcode == ENOMEM) caml_raise_out_of_memory();
-  err = strerror(retcode);
+  if (strerror_r(retcode, err, sizeof(err)) != 0) {
+    errlen = 0;
```

Comment:
I expect it's unlikely, but we get bitten by unexpected locale issues from time-to-time - could we handle `ERANGE` here, but just displaying a message that we failed to retrieve the description of the error?

Reply:
It really seems unlikely that we'll hit an ERANGE in the new implementation, since it has a 1024 TLS buffer (and that's well above anything libc produces, and glibc even documents it as an upper limit).  The actual handling of ERANGE varies a lot across platforms too (for example, macOS guarantees there will be a NUL-terminated truncated error, but some other platforms dont).